app/Http/Controllers/Api/V1/MasterSettingController.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\Http\Controllers\Api\V1;
+
+use App\Traits\ApiResponser;
+use Illuminate\Http\Request;
+use OpenApi\Attributes as OA;
+use App\Http\Controllers\Controller;
+use App\Services\MasterSettingService;
+use App\Http\Resources\MasterSetting\Resource;
+use App\Http\Resources\MasterSetting\Collection;
+
+class MasterSettingController extends Controller
+{
+    use ApiResponser;
+
+    private MasterSettingService $masterSettingService;
+
+    public function __construct()
+    {
+        $this->masterSettingService = new MasterSettingService;
+    }
+
+    #[OA\Get(
+        path: '/api/v1/versions',
+        operationId: 'getMasterSettings',
+        tags: ['MasterSettings'],
+        summary: 'Get list of settings',
+        responses: [
+            new OA\Response(
+                response: 200,
+                description: 'Success'
+            ),
+        ],
+        parameters: [
+            new OA\Parameter(
+                name: 'X-Requested-With',
+                in: 'header',
+                required: true,
+                description: 'Custom header for XMLHttpRequest',
+                schema: new OA\Schema(
+                    type: 'string',
+                    default: 'XMLHttpRequest'
+                )
+            ),
+        ],
+    )]
+    public function index(Request $request)
+    {
+        $masterSettings = $this->masterSettingService->collection($request->all());
+
+        return $this->collection(new Collection($masterSettings));
+    }
+
+    #[OA\Get(
+        path: '/api/v1/versions/{masterSetting_id}',
+        operationId: 'getVersionDetail',
+        tags: ['MasterSettings'],
+        summary: 'Get detail of settings',
+        responses: [
+            new OA\Response(
+                response: 200,
+                description: 'Success'
+            ),
+        ],
+        parameters: [
+            new OA\Parameter(
+                name: 'X-Requested-With',
+                in: 'header',
+                required: true,
+                description: 'Custom header for XMLHttpRequest',
+                schema: new OA\Schema(
+                    type: 'string',
+                    default: 'XMLHttpRequest'
+                )
+            ),
+            new OA\Parameter(
+                name: 'masterSetting_id',
+                in: 'path',
+                required: true,
+                description: 'Id of the setting',
+                schema: new OA\Schema(
+                    type: 'string'
+                )
+            ),
+        ],
+    )]
+    public function show($id)
+    {
+        $masterSettingObj = $this->masterSettingService->resource($id);
+
+        return isset($masterSettingObj['errors']) ? $this->error($masterSettingObj) : new Resource($masterSettingObj);
```

Comment:
@damini-7span If we have throw the error than we call the exception from the service file, So no need to check the error here.